### DISCUSSION TITLE
HBW-LC4-IN4-DR - Ausgänge werden nicht erkannt

### DISCUSSION THREAD

**Discussion by holzwurm83** [SukramJ/aiohomematic#1438] [Ideas]
Beim HBW-LC4-IN4-DR werden die 4 Ausgänge (Kanal 5-8) nicht erkannt. Je nachdem welche Version man gekauft hat. Werden Pro Ausgang 0-10 oder 1-10V ausgegeben. Das wir in der CCU unabhängig von der Version mit einem Dimmer 0-100% umgesetzt.  

Hier ein kompletter Auszug aus der CCU:
![image](https://github.com/danielperna84/hahomematic/assets/126335004/afcb27cc-8ce0-4650-a389-88146cec3d14)

Hier ein Auszug aus HomeAssistent:
Gerät 1:
![image](https://github.com/danielperna84/hahomematic/assets/126335004/2e891cc2-d5fd-4973-a1b7-e74b171ff4bd)
Gerät 2:
![image](https://github.com/danielperna84/hahomematic/assets/126335004/1f89c527-693d-4a2f-808c-5ce6f485f586)

Ich nutzte zwei dieser Geräte. Einmal um Dimmer über eine 0-10V Schnittstelle  zu Steuern und einmal für meine Lüftungsanlage die nur eine Schnittstelle mit 0-10 V verfügt. 

Können die Ausgänge  als Dimmer angelegt werden und der Verwendungszweck wie hier je nach Verwendungszweck definiert werden?

![image](https://github.com/danielperna84/hahomematic/assets/126335004/601c1e24-11a3-4c17-90c7-40bf698de4b3)

Hier ist der [Link](https://haus-bus.secure-stores.de/?showProduct=6) zur Seite des Herstellers.
Hier ist die dazugehörige xml-Datei
[hbw_lc4_in4_dr.xml.zip](https://github.com/danielperna84/hahomematic/files/14318481/hbw_lc4_in4_dr.xml.zip)

**Comment by SukramJ** [comment]
Mit der XML Kann ich nichts anfangen.
Biite den Service [homematicip_local.export_device_definition](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition), und die beiden Dateien als zip hier anhängen.

**Reply by holzwurm83** [reply]
Wie gewünscht...bitteschön! 
[export.zip](https://github.com/danielperna84/hahomematic/files/14319487/export.zip)

**Reply by SukramJ** [reply]
Das Einzige, was sich ändern wird, ist, dass die `number` Entitäten als `light` angezeigt werden.

**Reply by holzwurm83** [reply]
Hi, ich glaube das ich dir nicht folgen kann? Aktuell sieht das so aus:

![image](https://github.com/danielperna84/hahomematic/assets/126335004/b05e7e55-bddd-4486-864e-2105a771ec08)


Kann man das dann nicht wie hier auswählen? 
![305638505-601c1e24-11a3-4c17-90c7-40bf698de4b3](https://github.com/danielperna84/hahomematic/assets/126335004/38a2aec2-aa01-4511-89b4-c5d199ef9c8b)

**Reply by SukramJ** [reply]
Kann man auch.

**Reply by holzwurm83** [reply]
Denke das wäre zielführende. Ich ja nicht gesagt, dass es immer ein Licht sein wird.

**Reply by SukramJ** [reply]
Das Gerät wird als Dimmer beschrieben und sollte also in HA also `light` Entität erstellt werden.
Ob `light` oder `number` Entität ändert in diesem Fall nichts an der reinen Funktionalität, aber HA geht in der UI ggf. damit anders um.

Im nächsten Release wird das eine `light` Entität.

**Reply by SukramJ** [reply]
... und wem es nicht gefällt kann mit einem [ignore](https://github.com/danielperna84/hahomematic/blob/devel/docs/unignore.md#ignore) die Ursprungszustand wieder herstellen.

**Reply by holzwurm83** [reply]
Alles klar! 

> Im nächsten Release wird das eine light Entität.

Muss ich dann das bestehende Gerät wieder löschen?

**Reply by SukramJ** [reply]
Ja

**Comment by holzwurm83** [comment]
hi, ich habe gestern das update mit der aktuellen Version eingespielt nun sind alle Sensoren weg. 

<img width="330" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/a00aa5d1-097c-421f-b3c4-f0fa7d0792fc">

So, sah das vorher aus. Die Eingänge hatte ja gepasst. 

<img width="330" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/e3aa75bf-94ee-46b9-9b1e-ce6243d50f4a">

**Reply by SukramJ** [reply]
Dann lösch die Sensoren doch, die entsprechen doch den Ereignissen.

**Reply by holzwurm83** [reply]
das ist, aber so nicht richtig. Da stimmt was nicht. Es werden Taster erkannt erkannt die langen und kurzen Tastendruck anzeigen. Die Sensoren die offen und geschlossen anzeigen sind weg. 

Hier ein Auszug aus der CCU:

<img width="1038" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/21d0f6cb-47f0-415f-be67-4962d5402f0e">

**Reply by SukramJ** [reply]
Pack dir einfach SENSOR in [unignore](https://github.com/danielperna84/custom_homematic#unignore-device-parameters)

**Reply by holzwurm83** [reply]
wie muss die Zeile genau aussehen? 

SENSOR:hbw_lc4_in4_dr_hbw0001980

**Reply by SukramJ** [reply]
Hast du wirklich gelesen?

`SENSOR`
oder
```
SENSOR:VALUES@HBW-LC4-IN4-DR:1
...
```

**Reply by holzwurm83** [reply]
gelesen schon nur das mit dem verstehen ist meinst nicht so 😔 Sorry!

was mir jetzt noch nicht klar ist:

- Ich habe zwei Geräte von HBW-LC4-IN4-DR und wie kann ich dann dazu unterscheiden auf Kanalebene
- einen von HBW-LC4-IN4-DR wollte ich mit ignore_HBW-LC4-IN4-DR raus nehmen, aber nur einen und nicht beide

**Reply by SukramJ** [reply]
Das geht nicht.

**Reply by holzwurm83** [reply]
ich wollte das so probieren:

SENSOR:VALUES@HBW-LC4-IN4-DR:HBW0001484:1
ignore_HBW-LC4-IN4-DR:HBW0001484


wenn das alles nicht stehen wir wieder leider wieder am Anfang. 

Ein Gerät ist die Lüftung angeschlossen und am zweiten hängen die Dimmer. 

Können die Ausgänge als Dimmer angelegt werden und der Verwendungszweck wie hier je nach Verwendungszweck definiert werden?

[image](https://private-user-images.githubusercontent.com/126335004/305638505-601c1e24-11a3-4c17-90c7-40bf698de4b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5ODI5NDIsIm5iZiI6MTcwOTk4MjY0MiwicGF0aCI6Ii8xMjYzMzUwMDQvMzA1NjM4NTA1LTYwMWMxZTI0LTExYTMtNGMxNy05MGM3LTQwYmY2OThkZTRiMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwOVQxMTEwNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOWM5MWQzNjNkMWVjYWFmM2E5ZGI5MmIyNjM0ODU5ZjY0MDdmOWRmYTc5MzhkYTAzMzllY2E1YjM5ZTMxMDgyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BWDyyDLivC73h9984R446QWZ0jAKdyhPFhv1m5Vpk0s)

**Reply by SukramJ** [reply]
Sich irgendwas auszudenken ist .....

> wenn das alles nicht stehen wir wieder leider wieder am Anfang.

Wir nicht!

Einfach ignore_HBW-LC4-IN4-DR in die unignore Datei reinschreiben, und danach kannst Du die Geräte nach Herzenslust mit HA Templates definieren. 

Ob du da ein Lampe oder einen Lüfter angeschlossen hast ist HA ziemlich egal, im Zweifel kannst Du das Icon ändern.
Wenn Du solche Änderungen machen willst am Besten die HA Doku lesen, denn dafür ist HA da. Eine Integration kann dir deine Wunsch-Umgebung nicht auf dem Silbertablett servieren.
Ich werde da nichts weiter machen.

**Reply by holzwurm83** [reply]
hey, tut mir leid. Meine bitte war es einfach nur, das die Darstellung aus der CCU übernommen wird und das Gerät für die Nachwelt richtig erkannt wird. 

Ich verstehe nicht warum ignore_HBW-LC4-IN4-DR dann nur pauschal funktioniert? Das setzt voraus, dass man dann jedes Gerät nur einmal haben darf und wenn ein Gerät anders verwendet wird müssen alle zurückgesetzt und grundlegend konfiguriert werden. Der Aufwand ist dafür sehr groß und der weg für die meisten sehr steinig und führt nicht immer zum erfolg. 😔

**Reply by SukramJ** [reply]
> Der Aufwand ist dafür sehr groß und der weg für die meisten sehr steinig und führt nicht immer zum erfolg. 

Jetzt muss ich wirklich lachen. Weist Du wieviel hunderte Stunden ich investiert habe?

ignore_HBW-LC4-IN4-DR ist eine Variante, die man nutzen kann, aber nicht muss! In deinem Fall reicht es vielleicht einfach das Icon zu ändern.
